### PR TITLE
fix(memo-timeout-time):  use u64 instead of string

### PR DESCRIPTION
**Pull request by RustNinja**
Required for merge:
- [x] `pr-workflow-check / draft-release-check` is ✅ success
- Other rules GitHub shows you, or can be read in [configuration](../terraform/github.com/branches.tf) 

Makes review faster:
- [x] PR title is my best effort to provide summary of changes and has clear text to be part of release notes 
- [x] I marked PR by `misc` label if it should not be in release notes
- [x] Linked Zenhub/Github/Slack/etc reference if one exists
- [x] I was clear on what type of deployment required to release my changes (node, runtime, contract, indexer, on chain operation, frontend, infrastructure) if any in PR title or description
- [x] Added reviewer into `Reviewers`
- [x] I tagged(`@`) or used other form of notification of one person who I think can handle best review of this PR
- [x] I have proved that PR has no general regressions of relevant features and processes required to release into production
- [x] Any dependency updates made, was done according guides from relevant dependency
- Clicking all checkboxes 
- Adding detailed description of changes when it feels appropriate (for example when PR is big)

### CODE REVIEW

File: code/xcvm/lib/core/src/transport/ibc/ics20/pfm.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ pub struct ForwardingMemo {
 	#[serde(skip_serializing_if = "Option::is_none")]
 	pub channel: Option<ChannelId>,
 	#[serde(skip_serializing_if = "Option::is_none")]
-	pub timeout: Option<String>,
+	pub timeout: Option<u64>,
```
Comment (mina86) [inline code/xcvm/lib/core/src/transport/ibc/ics20/pfm.rs:18]: Note that serde uses integer representation for u64.  Serde talking to serde will work fine but we may run into problems if this memo would be ever read by JavaScript for example.  Perhaps better use `Displayed<u64>`.

**Comment by dzmitry-lahoda** [comment]
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:30
composable-deps>    |
composable-deps> 10 |     __m128, __m128i, __m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph,
composable-deps>    |                              ^^^^^^^^^^^^^^^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:47
composable-deps>    |
composable-deps> 10 |     __m128, __m128i, __m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph,
composable-deps>    |                                               ^^^^^^^^^^^^^^^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:64
composable-deps>    |
composable-deps> 10 |     __m128, __m128i, __m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph,
composable-deps>    |                                                                ^^^^^^^^^^^^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:78
composable-deps>    |
composable-deps> 10 |     __m128, __m128i, __m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph,
composable-deps>    |                                                                              ^^^^^^^^^^^^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:26:18
composable-deps>    |
composable-deps> 26 |     let retval = _mm_cvtph_ps(vec.assume_init());
composable-deps>    |                  ^^^^^^^^^^^^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/lib.rs:1:1
composable-deps>    |
composable-deps> 1  | / //! A crate that provides support for half-precision 16-bit floating point types.
composable-deps> 2  | | //!
composable-deps> 3  | | //! This crate provides the [`f16`] type, which is an implementation of the IEEE 754-2008 standard
composable-deps> 4  | | //! [`binary16`] a.k.a "half" floating point type. This 16-bit floating point type is intended for
composable-deps> ...  |
composable-deps> 34 | | //! A [`prelude`] module is provided for easy importing of available utility traits.
composable-deps> 35 | | //!
composable-deps>    | |_____________________________^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:44:18
composable-deps>    |
composable-deps> 44 |     let retval = _mm_cvtph_ps(vec.assume_init());
composable-deps>    |                  ^^^^^^^^^^^^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/lib.rs:1:1
composable-deps>    |
composable-deps> 1  | / //! A crate that provides support for half-precision 16-bit floating point types.
composable-deps> 2  | | //!
composable-deps> 3  | | //! This crate provides the [`f16`] type, which is an implementation of the IEEE 754-2008 standard
composable-deps> 4  | | //! [`binary16`] a.k.a "half" floating point type. This 16-bit floating point type is intended for
composable-deps> ...  |
composable-deps> 52 | | //! # Hardware support
composable-deps> 53 | | //!
composable-deps>    | |_____________________________^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:85:18
composable-deps>    |
composable-deps> 85 |     let retval = _mm256_cvtph_ps(vec.assume_init());
composable-deps>    |                  ^^^^^^^^^^^^^^^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps> error[E0658]: use of unstable library feature 'stdsimd'
composable-deps>   --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/lib.rs:1:1
composable-deps>    |
composable-deps> 1  | / //! A crate that provides support for half-precision 16-bit floating point types.
composable-deps> 2  | | //!
composable-deps> 3  | | //! This crate provides the [`f16`] type, which is an implementation of the IEEE 754-2008 standard
composable-deps> 4  | | //! [`binary16`] a.k.a "half" floating point type. This 16-bit floating point type is intended for
composable-deps> ...  |
composable-deps> 93 | | //!
composable-deps> 94 | | //! - **`bytemuck`** — Adds support for the [`bytemuck`] crate by implementing [`Zeroable`] and
composable-deps>    | |________________________________^
composable-deps>    |
composable-deps>    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
composable-deps>    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
composable-deps>    Compiling cxx-build v1.0.104
composable-deps>    Compiling bech32 v0.9.1
composable-deps>    Compiling safe-quote v1.0.15
composable-deps> For more information about this error, try `rustc --explain E0658`.
composable-deps> error: could not compile `half` (lib) due to 10 previous errors
composable-deps> warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/qprfsckl8fjbkwkdahq23s8c8gq2gbay-composable-deps-8.10035.2.drv' failed with exit code 101;
       last 10 log lines:
       >    |
       >    = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
       >    = help: add `#![feature(stdsimd)]` to the crate attributes to enable
       >
       >    Compiling cxx-build v1.0.104
       >    Compiling bech32 v0.9.1
       >    Compiling safe-quote v1.0.15
       > For more information about this error, try `rustc --explain E0658`.

**Comment by dzmitry-lahoda** [comment]
typicall error when std dep was added to no std build and cannot compile runtime

**Comment by RustNinja** [comment]
> 

it is very strange:
cargo build -p picasso-runtime-wasm    --target wasm32-unknown-unknown
cargo build -p composable-runtime-wasm    --target wasm32-unknown-unknown

compiles runtime into wasm successfully

**Comment by RustNinja** [comment]
> composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:30 composable-deps> | composable-deps> 10 | **m128, m128i, m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph, composable-deps> | ^^^^^^^^^^^^^^^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:47 composable-deps> | composable-deps> 10 | m128, m128i, m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph, composable-deps> | ^^^^^^^^^^^^^^^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:64 composable-deps> | composable-deps> 10 | m128, m128i, m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph, composable-deps> | ^^^^^^^^^^^^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:10:78 composable-deps> | composable-deps> 10 | m128, m128i, m256, _mm256_cvtph_ps, _mm256_cvtps_ph, _mm_cvtph_ps, _mm_cvtps_ph, composable-deps> | ^^^^^^^^^^^^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:26:18 composable-deps> | composable-deps> 26 | let retval = _mm_cvtph_ps(vec.assume_init()); composable-deps> | ^^^^^^^^^^^^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/lib.rs:1:1 composable-deps> | composable-deps> 1 | / //! A crate that provides support for half-precision 16-bit floating point types. composable-deps> 2 | | //! composable-deps> 3 | | //! This crate provides the [`f16`] type, which is an implementation of the IEEE 754-2008 standard composable-deps> 4 | | //! [`binary16`] a.k.a "half" floating point type. This 16-bit floating point type is intended for composable-deps> ... | composable-deps> 34 | | //! A [`prelude`] module is provided for easy importing of available utility traits. composable-deps> 35 | | //! composable-deps> | |**_**^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:44:18 composable-deps> | composable-deps> 44 | let retval = _mm_cvtph_ps(vec.assume_init()); composable-deps> | ^^^^^^^^^^^^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/lib.rs:1:1 composable-deps> | composable-deps> 1 | / //! A crate that provides support for half-precision 16-bit floating point types. composable-deps> 2 | | //! composable-deps> 3 | | //! This crate provides the [`f16`] type, which is an implementation of the IEEE 754-2008 standard composable-deps> 4 | | //! [`binary16`] a.k.a "half" floating point type. This 16-bit floating point type is intended for composable-deps> ... | composable-deps> 52 | | //! # Hardware support composable-deps> 53 | | //! composable-deps> | |**_**^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/binary16/arch/x86.rs:85:18 composable-deps> | composable-deps> 85 | let retval = _mm256_cvtph_ps(vec.assume_init()); composable-deps> | ^^^^^^^^^^^^^^^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> error[E0658]: use of unstable library feature 'stdsimd' composable-deps> --> /nix/store/zcd5ww170nk1hg3ibmadpbf9a54jv31f-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/half-2.3.1/src/lib.rs:1:1 composable-deps> | composable-deps> 1 | / //! A crate that provides support for half-precision 16-bit floating point types. composable-deps> 2 | | //! composable-deps> 3 | | //! This crate provides the [`f16`] type, which is an implementation of the IEEE 754-2008 standard composable-deps> 4 | | //! [`binary16`] a.k.a "half" floating point type. This 16-bit floating point type is intended for composable-deps> ... | composable-deps> 93 | | //! composable-deps> 94 | | //! - `bytemuck` — Adds support for the [`bytemuck`] crate by implementing [`Zeroable`] and composable-deps> | |**________^ composable-deps> | composable-deps> = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information composable-deps> = help: add `#![feature(stdsimd)]` to the crate attributes to enable composable-deps> Compiling cxx-build v1.0.104 composable-deps> Compiling bech32 v0.9.1 composable-deps> Compiling safe-quote v1.0.15 composable-deps> For more information about this error, try `rustc --explain E0658`. composable-deps> error: could not compile `half` (lib) due to 10 previous errors composable-deps> warning: build failed, waiting for other jobs to finish... error: builder for '/nix/store/qprfsckl8fjbkwkdahq23s8c8gq2gbay-composable-deps-8.10035.2.drv' failed with exit code 101; last 10 log lines: > | > = note: see issue #48556 [rust-lang/rust#48556](https://github.com/rust-lang/rust/issues/48556) for more information > = help: add `#![feature(stdsimd)]` to the crate attributes to enable > > Compiling cxx-build v1.0.104 > Compiling bech32 v0.9.1 > Compiling safe-quote v1.0.15 > For more information about this error, try `rustc --explain E0658`.

does it mean that i am not able to use u64 in serde???

File: code/xcvm/lib/core/src/transport/ibc/ics20/pfm.rs (hunk 1 above)
Comment (RustNinja) [reply code/xcvm/lib/core/src/transport/ibc/ics20/pfm.rs:18]: pub height: Option<u64>,
works fine so i just replaced String with u64 for other field.

**Comment by mina86** [comment]
You should be able to use it. It’s more likely that the Cargo.toml updates you’ve made broke the build. Split this PR into two — one updating the dependencies and the other changing type of timeout — to check.

File: code/parachain/frame/composable-traits/src/centauri.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@ impl Map {
 				receiver: value.receiver,
 				port: value.port.map(|x| x.to_string()),
 				channel: value.channel.map(|x| x.to_string()),
-				timeout: value.timeout,
+				timeout: value.timeout.and_then(|x| x.parse().ok()),
```
Comment (mina86) [inline code/parachain/frame/composable-traits/src/centauri.rs:16]: This error should propagate up.

**Comment by RustNinja** [comment]
closed because created other pr
https://github.com/ComposableFi/composable/pull/4049#event-10059601112

**Pull request closed without merging**